src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java

```diff
@@ -54,4 +60,32 @@ public LogGroupsResponse getLogGroupNames(String region, String awsAccessKeyId,
         LOG.debug("Log groups queried: [{}]", groupNameList);
         return LogGroupsResponse.create(groupNameList, groupNameList.size());
     }
+
+    public String addSubscriptionFilter(CloudWatchLogsClient cloudWatch, String logGroup, String streamArn,
+                                        String roleArn, String filterName,
+                                        String filterPattern) throws InterruptedException {
+        PutSubscriptionFilterRequest putSubscriptionFilterRequest =
+                PutSubscriptionFilterRequest.builder()
+                                            .logGroupName(logGroup)
+                                            .filterName(filterName)
+                                            .filterPattern(filterPattern)
+                                            .destinationArn(streamArn)
+                                            .roleArn(roleArn)
+                                            .distribution(Distribution.BY_LOG_STREAM)
+                                            .build();
+        sleep(2_000);
+        try {
+            cloudWatch.putSubscriptionFilter(putSubscriptionFilterRequest);
+            final String responseMessage = String.format("Success. The subscription filter [%s] was added to [%s].",
+                                                         filterName, logGroup);
+            return responseMessage;
+        } catch (Exception e) {
+            final String specificError = ExceptionUtils.formatMessageCause(e);
+            final String responseMessage = String.format("Attempt to add subscription [%s] to Cloudwatch log group " +
```

Comment:
This `responseMessage` variable is not needed and can be in inline return statement like this:
```suggestion
            return String.format("Attempt to add subscription [%s] to Cloudwatch log group " +
```

Reply:
I kept the responseMessage and instead included `throw new BadRequestException(responseMessage, e);`